frame/bounties/src/lib.rs

```diff
@@ -653,12 +780,596 @@ decl_module! {
 		}
+
+		/// Add a new subbounty.
+		///
+		/// The dispatch origin for this call must be curator.
+		/// Bounty must me in "active" state.
+		///
+		/// Subbouty gets added successfully & fund gets reserved, if bounty has enough fund.
+		/// else call get failed.
+		///
+		/// Upperbount to maximum number of subbounties that can be added is
+		/// managed via runtime trait config 'MaxSubBountyCount'.
+		///
+		/// Payment: `TipReportDepositBase` will be reserved from the origin account, as well as
+		/// `DataDepositPerByte` for each byte in `reason`. It will be unreserved upon approval,
+		/// or slashed when rejected.
+		///
+		/// if call is success, state of subbounty is moved to "Approved" state.
+		/// And later moved to "Funded" state as part of "spend_fund()" callback.
+		///
+		/// - `bounty_id`: Bounty ID for which subbounty to be added.
+		/// - `value`: Value for executing the proposal.
+		/// - `description`: Text description for the subbounty.
+		#[weight = 10_000]
+		fn add_subbounty(origin,
+			#[compact] bounty_id: BountyIndex,
+			value: BalanceOf<T>,
+			description: Vec<u8>,
+		)
+		{
+			let signer = ensure_signed(origin)?;
+
+			Bounties::<T>::try_mutate_exists(bounty_id, |maybe_bounty| -> DispatchResult {
+				let mut bounty = maybe_bounty.as_mut().ok_or(Error::<T>::InvalidIndex)?;
+				match bounty.status {
+					BountyStatus::Active {
+						ref curator,
+						..
+					} => {
+						ensure!(signer == *curator, Error::<T>::RequireCurator);
+						// Verify the arguments
+						ensure!(description.len() <=
+							T::MaximumReasonLength::get() as usize,
+							Error::<T>::ReasonTooBig,
+						);
+						ensure!(value >= T::BountyValueMinimum::get(),
+							Error::<T>::InvalidValue,
+						);
+						ensure!((bounty.active_subbounty.len() as u32) <
+							T::MaxSubBountyCount::get() as u32,
+							Error::<T>::SubBountyMaxOverflow,
+						);
+
+						// Makesure Parent bounty have enough balance to fund Subbounty
+
+						// Adding first subbounty, update the state of subbounty
+						// free balance tracking.
+						if bounty.subbounty_count == 0 {
+							let bounty_account = Self::bounty_account_id(bounty_id);
+							let balance = T::Currency::free_balance(&bounty_account);
+							// reserve the fund for master curator fee
+							bounty.subbounty_free_balance = balance.saturating_sub(bounty.fee);
+						}
+
+						let tst_new_free_balance = bounty.subbounty_free_balance.saturating_sub(value);
+
+						ensure!(tst_new_free_balance > Zero::zero(), Error::<T>::InsufficientBountyBalance);
+
+						// Increment the subbounty tracking count
+						// And i
+						bounty.subbounty_count += 1;
+						match bounty.active_subbounty.binary_search_by_key(
+							&bounty.subbounty_count,
+							|x| *x,
+						) {
+							Ok(_) => {
+								//This should not occur
+							},
+							Err(i) => bounty.active_subbounty.insert(i, bounty.subbounty_count),
+						}
+
+						let _ = Self::create_subbounty(
```

Comment:
never ignore error without explain why. in this case `create_subbounty` can't fail so don't make it return `Result`

Reply:
Refactored the code, return value from "create_subbounty()" removed.